src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -85,13 +123,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         return;
     }
 
+    private Entity createMealEntity(Meal meal) {
+        Entity mealEntity = new Entity("Meal");
+        mealEntity.setProperty("id", meal.getId());
+        mealEntity.setProperty("title", meal.getTitle());
+        mealEntity.setProperty("description", meal.getDescription());
+        mealEntity.setProperty("ingredients", meal.getIngredients());
+        mealEntity.setProperty("type", meal.getType());
+ 
+        return mealEntity;
+    }
+
+    private void getMealList(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Meal").addSort("id", SortDirection.DESCENDING);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Meal> meals = getDataFromDatastore(results);
+        Gson gson = new Gson();
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(meals));
+        return;
+    }
+
     private void searchMeal(HttpServletRequest request, HttpServletResponse response) throws IOException {
         // Fuction returns hardcoded result
         // TODO(sandatsian): implement search algorithm here for MVP
-        String gson = new Gson().toJson(dishes);
+        /*String gson = new Gson().toJson(dishes);
         response.setContentType("application/json");
         response.getWriter().println(gson);
-        return;
+        return;*/
     }
 
     private void getMealById(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
I need you to write unit tests for these features - getting a single element by ID and listing all the elements.

In these tests you need to create an instance of the MealServlet class, and then either call the `doGet()` method in the test, or expose the `getMealById` and `getMealList` methods so you can call them in the test.

Since your tests aren't supposed to access the real Datastore in GCE, you'll need to follow this [Local unit testing for Java 8](https://cloud.google.com/appengine/docs/standard/java/tools/localunittesting) AppEngine tutorial. Each test will need to prepare the database for the tested scenario (by configuring the service with `LocalDatastoreServiceTestConfig`, and then adding some hardcoded examples as a part of test's code).